fenix/app/src/main/java/org/mozilla/fenix/library/history/state/HistoryUiEffectMiddleware.kt

```diff
@@ -0,0 +1,35 @@
+package org.mozilla.fenix.library.history.state
+
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+import org.mozilla.fenix.library.history.HistoryFragmentAction
+import org.mozilla.fenix.library.history.HistoryFragmentState
+
+class HistoryUiEffectMiddleware(
```

Comment:
it feels a little weird to have a middleware just to trigger UI side effects that could just be a part of their state observation, but that is undoubtedly an implementation issue with History.

Reply:
This was meant to demo callback handling through middleware. I think it many cases it would be more appropriate to launch a side-effect from observing a change to state, but I also don't think it is necessary to track things like `isMenuOpen` in the state for a screen. For example if we have the following action:

```
sealed class ScreenAction {
    data class ItemLongClicked(val item: Item) : ScreenAction
}
```

Handling the side-effect from a middleware makes more sense to me
```
class ScreenUiMiddleware(
    private val displayItemMenu: (Item) -> Unit
)  : Middleware<ScreenState, ScreenAction> {
    override fun invoke(
        context: MiddlewareContext<ScreenState, ScreenAction>,
        next: (ScreenAction) -> Unit,
        action: ScreenAction,
    ) {
        when (action) {
             is ItemLongClicked -> displayItemMenu(action.item)
         }
    }
}
```